crates/nu-cli/src/shell/palette.rs

```diff
@@ -125,6 +135,194 @@ impl Palette for ThemedPallet {
+    #[serde(with = "string_and_color")]
+    external_command: Color,
+    #[serde(with = "string_and_color")]
+    external_word: Color,
+    #[serde(with = "string_and_color")]
+    bare_member: Color,
+    #[serde(with = "string_and_color")]
+    string_member: Color,
+    #[serde(with = "string_and_color")]
+    string: Color,
+    #[serde(with = "string_and_color")]
+    path: Color,
+    #[serde(with = "string_and_color")]
+    word: Color,
+    #[serde(with = "string_and_color")]
+    keyword: Color,
+    #[serde(with = "string_and_color")]
+    pipe: Color,
+    #[serde(with = "string_and_color")]
+    glob_pattern: Color,
+    #[serde(with = "string_and_color")]
+    flag: Color,
+    #[serde(with = "string_and_color")]
+    shorthand_flag: Color,
+    #[serde(with = "string_and_color")]
+    int: Color,
+    #[serde(with = "string_and_color")]
+    decimal: Color,
+    #[serde(with = "string_and_color")]
+    garbage: Color,
+    #[serde(with = "string_and_color")]
+    whitespace: Color,
+    #[serde(with = "string_and_color")]
+    separator: Color,
+    #[serde(with = "string_and_color")]
+    comment: Color,
+    #[serde(with = "string_and_color")]
+    size_number: Color,
+    #[serde(with = "string_and_color")]
+    size_unit: Color,
+}
+
+mod string_and_color {
+    use ansi_term::Color;
+    use serde::{self, Deserialize, Deserializer, Serializer};
+    use std::str::Bytes;
+
+    pub fn serialize<S>(color: &Color, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str("TODO: IMPLEMENT SERIALIZATION")
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Color, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        to_color(&s)
+    }
+
+    fn to_color<E>(s: &str) -> Result<Color, E>
+    where
+        E: serde::de::Error,
+    {
+        let mut bytes = s.bytes();
+        let r = xtoi(&mut bytes)?;
+        let g = xtoi(&mut bytes)?;
+        let b = xtoi(&mut bytes)?;
+        Ok(Color::RGB(r, g, b))
+    }
+
+    fn xtoi<E>(b: &mut Bytes) -> Result<u8, E>
+    where
+        E: serde::de::Error,
+    {
+        let upper = b.next().ok_or(E::custom("color string too short"))?;
+        let lower = b.next().ok_or(E::custom("color string too short"))?;
+        let mut val = numerical_value(upper)?;
+        val = (val << 4) | numerical_value(lower)?;
+        Ok(val)
+    }
+
+    fn numerical_value<E>(character: u8) -> Result<u8, E>
+    where
+        E: serde::de::Error,
+    {
+        match character {
+            b'0'..=b'9' => Ok(character - b'0'),
+            b'a'..=b'z' => Ok(character - (b'a' - 10)),
+            _ => return Err(E::custom(format!("invalid charater {}", character))),
+        }
+    }
+}
+
 fn single_style_span(style: Style, span: Span) -> Vec<Spanned<Style>> {
     vec![Spanned::<Style> { span, item: style }]
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Is this where my tests should live? I poked around a little bit but I couldn't find a place to put this test that I felt good about.

Reply:
If they're unit tests, you can put them beside the code like this.